InkscapeExtension/ExtendedHatchFill.py

```diff
@@ -96,8 +101,109 @@ def recursivelyTraverseSvg(self, a_node_list):
                     self.interstices((h[0], h[1]), (h[2], h[3]))
 
     def buildHatchLinePath(self, pt1, pt2):
-        return 'M {0:f},{1:f} l {2:f},{3:f} '.format(pt1[0], pt1[1], pt2[0] - pt1[0], pt2[1] - pt1[1])
+         x1,y1 = pt1[0],pt1[1]
+         x2,y2 = pt2[0],pt2[1]
 
+         od = self.options.outer_dist # Outer contour indent
+         cc = self.options.curv_coef # Curvature coefficient (steepness angle)                   
+         angle = math.radians(self.options.hatchAngle)
+
+          # absolute length of the segment between the points
+         totLen = ((x2 - x1) ** 2 + (y2 - y1) ** 2) ** 0.5          
+         w = x2 - x1 
+         h = y2 - y1 
+         wx = w / totLen # * math.cos(math.radians((angle)) # cos
+         hx = h / totLen # * math.sin(math.radians((angle)) # sin
+        
+         r1 = randint(-cc, cc)  
+         r2 = randint(-cc, cc)
+         r3 = randint(-cc, cc)
+         r4 = randint(-cc, cc)
+
+
+         if self.options.mode == "linear":
+            path = f'M {x1 + r1 + od * wx}, {y1 + r2 + od * hx} l {x2-x1 -r3 - 2 * od * wx}, {y2-y1 - r4 - 2 * od * hx}'
+         
+         if self.options.mode == "quadratic":
```

Comment:
I think it makes sense to replace 2nd and following "if"s with "elif". Do you agree?

Reply:
Yes, of course!